prov/gni/src/gnix_rma.c

```diff
@@ -327,18 +626,27 @@ int _gnix_rma_post_req(void *data)
 				(gnix_mr_key_t *)&fab_req->rma.rem_mr_key,
 				&mdh);
 	}
-	loc_md = (struct gnix_fid_mem_desc *)fab_req->rma.loc_md;
 
 	txd->gni_desc.type = __gnix_fr_post_type(fab_req->type, rdma);
 	txd->gni_desc.cq_mode = GNI_CQMODE_GLOBAL_EVENT; /* check flags */
 	txd->gni_desc.dlvr_mode = GNI_DLVMODE_PERFORMANCE; /* check flags */
-	txd->gni_desc.local_addr = (uint64_t)fab_req->rma.loc_addr;
-	if (loc_md) {
-		txd->gni_desc.local_mem_hndl = loc_md->mem_hndl;
+
+	if (indirect) {
+		__gnix_rma_fill_pd_indirect_get(fab_req, txd);
+	} else if (chained) {
+		__gnix_rma_fill_pd_chained_get(fab_req, txd, &mdh);
+	} else {
+		txd->gni_desc.local_addr = (uint64_t)fab_req->rma.loc_addr;
+		txd->gni_desc.length = fab_req->rma.len;
+		txd->gni_desc.remote_addr = (uint64_t)fab_req->rma.rem_addr;
+
+		loc_md = (struct gnix_fid_mem_desc *)fab_req->rma.loc_md;
+		if (loc_md) {
+			txd->gni_desc.local_mem_hndl = loc_md->mem_hndl;
+		}
```

Comment:
I think it would be better from a performance perspective to at least use unlikely macro here for case of indirect/chained.